core/src/main/scala/dagr/core/execsystem/ResourceSet.scala

```diff
@@ -24,7 +24,7 @@
 package dagr.core.execsystem
 
 object ResourceSet {
-  def apply(that: ResourceSet): ResourceSet = new ResourceSet(cores = that.cores, memory = that.memory)
+  def apply(that: ResourceSet): ResourceSet = new ResourceSet(cores = Cores(that.cores.cores), memory = Memory(that.memory.value))
```

Comment:
Given that ResourceSet is immutable, perhaps it just makes sense to eliminate this apply method entirely?